src/main/java/su/litvak/chromecast/api/v2/Channel.java

```diff
@@ -173,6 +175,18 @@ public void run() {
                     }
                 } catch (InvalidProtocolBufferException ipbe) {
                     LOG.debug("Error while processing protobuf: {}", ipbe.getLocalizedMessage());
+                } catch (JsonParseException jpe) {
+                    LOG.warn("Error while processing protobuf: {}", jpe.getLocalizedMessage());
+                } catch (SocketException se) {
+                    LOG.warn("Socket error, will reconnect: {}", se.getLocalizedMessage());
+                    LOG.debug("StackTrace", se);
+                    try {
+                        close();
```

Comment:
Reconnection should only happen if the `autoReconnect` property is set to `true` in `ChromeCast` class. Thus, it makes sense to close this Channel on `SocketException` and leave reconnection to the upper level objects such as `ChromeCast`.